pyrelational/data/data_manager.py

```diff
@@ -121,85 +145,58 @@ def _ensure_no_l_or_u_leaks(self) -> None:
+    def _resolve_dataset_split_indices(
+        self,
+        train_indices: Optional[List[int]],
+        validation_indices: Optional[List[int]],
+        test_indices: Optional[List[int]],
+    ) -> None:
         """This function is used to resolve what values the indices should be given
         when only a partial subset of them is supplied
         """
-        train_flag = self.train_indices is not None
-        valid_flag = self.validation_indices is not None
-        test_flag = self.test_indices is not None
-
-        # Different cases for presence of training mask, validation mask, test mask
-        # TTT Case 0: All masks supplied; check for overlaps to avoid leaks
-        if train_flag and valid_flag and test_flag:
-            pass
-
-        # TTF Case 1: Any remaining samples become test
-        elif train_flag and valid_flag and not test_flag:
-            remaining_indices = set(range(len(self.dataset))) - set.union(
-                set(self.train_indices), set(self.validation_indices)
-            )
-            self.test_indices = list(remaining_indices)
-
-        # TFT Case 2: No validation, set any remaining as train (labelled and unlabelled handled seperately)
-        elif train_flag and not valid_flag and test_flag:
-            remaining_indices = set(range(len(self.dataset))) - set.union(
-                set(self.train_indices), set(self.test_indices)
-            )
-            self.train_indices = set.union(set(self.train_indices), remaining_indices)
-            self.train_indices = list(self.train_indices)
-
-        # TFF Case 3: Only train, set others as test
-        elif train_flag and not valid_flag and not test_flag:
-            remaining_indices = set(range(len(self.dataset))) - set(self.train_indices)
-            self.test_indices = list(remaining_indices)
-
-        # FTT Case 4: No train, set any remaining as train
-        elif not train_flag and valid_flag and test_flag:
-            remaining_indices = set(range(len(self.dataset))) - set.union(
-                set(self.validation_indices), set(self.test_indices)
-            )
-            self.train_indices = list(remaining_indices)
-
-        # FTF Case 5: Only validation, send an error
-        elif not train_flag and valid_flag and not test_flag:
-            raise ValueError("No train or test specified, too ambigious to set values")
-
-        # FFT Case 6: Only test, send an error
-        elif not train_flag and not valid_flag and test_flag:
-            raise ValueError("No train or validation specified, too ambigious to set values")
-
-        # FFF Case 7: No masks given, everything is a training observation
-        elif not train_flag and not valid_flag and not test_flag:
-            self.train_indices = list(range(len(self.dataset)))
 
-        else:
-            raise ValueError("Invalid split indices provided")
+        remaining_indices = set(range(len(self.dataset))) - set.union(
```

Comment:
looks a lot simpler, I like it! It may be useful to detail the strategy used in the docstring though